sda-commons-client-jersey/src/main/java/org/sdase/commons/client/jersey/oidc/rest/IssuerClient.java

```diff
@@ -0,0 +1,96 @@
+package org.sdase.commons.client.jersey.oidc.rest;
+
+import static org.sdase.commons.client.jersey.proxy.ApiClientInvocationHandler.createProxy;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.core.Form;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import org.glassfish.jersey.client.proxy.WebResourceFactory;
+import org.sdase.commons.client.jersey.ClientFactory;
+import org.sdase.commons.client.jersey.oidc.OidcConfiguration;
+import org.sdase.commons.client.jersey.oidc.rest.model.OpenIdDiscoveryResource;
+import org.sdase.commons.client.jersey.oidc.rest.model.TokenResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IssuerClient {
+  private static final Logger LOGGER = LoggerFactory.getLogger(IssuerClient.class);
+
+  private final OidcConfiguration config;
+  private final Client client;
+  private final OpenIdDiscoveryApi discoveryApi;
+
+  public IssuerClient(ClientFactory clientFactory, OidcConfiguration config) {
+    this.config = config;
+
+    this.client =
+        clientFactory.externalClient(config.getHttpClient()).buildGenericClient("oidc-client");
+
+    this.discoveryApi =
+        createProxy(
+            OpenIdDiscoveryApi.class,
+            WebResourceFactory.newResource(
+                OpenIdDiscoveryApi.class, client.target(config.getIssuerUrl())));
+  }
+
+  /**
+   * Retrieves a TokenResource from the configured OIDC provider.
+   *
+   * @return A TokenResource containing the accessToken, tokenType and its expirationTime.
+   */
+  public TokenResource getTokenResource() {
+    OpenIdDiscoveryResource discoveryResource = discoveryApi.getConfiguration();
+
+    if (discoveryResource == null || discoveryResource.getTokenEndpoint() == null) {
+      throw new IllegalStateException("Error loading discovery configuration");
```

Comment:
Have you tested the behaviour? I'd thought we'd just return `null` because that return type is already handled as ERROR state. What happens if you throw the exception?

Reply:
Right, this way its somewhat inconsistent as sometimes we are throwing exceptions and sometimes setting the `ERROR` state. I'll simply return null here to keep it consistent.